core/base/marchingTetrahedra/MarchingTetrahedra.h

```diff
@@ -0,0 +1,3603 @@
+
+        // Get the tetrahedron local vertex ids that all three have the same
+        // label
+        const int id0 = tetLookupFastTri[tetLookupFastCase[tetCases[tet]]][0];
+        const int id1 = tetLookupFastTri[tetLookupFastCase[tetCases[tet]]][1];
+        const int id2 = tetLookupFastTri[tetLookupFastCase[tetCases[tet]]][2];
+
+        float vertPos[4][3];
+        triangulation.getVertexPoint(
+          vertices[0], vertPos[0][0], vertPos[0][1], vertPos[0][2]);
+        triangulation.getVertexPoint(
+          vertices[1], vertPos[1][0], vertPos[1][1], vertPos[1][2]);
+        triangulation.getVertexPoint(
+          vertices[2], vertPos[2][0], vertPos[2][1], vertPos[2][2]);
+        triangulation.getVertexPoint(
+          vertices[3], vertPos[3][0], vertPos[3][1], vertPos[3][2]);
+
+        // Write the triangle endpoints, cell indices, and label
+        p[0] = vertPos[id0][0];
+        p[1] = vertPos[id0][1];
+        p[2] = vertPos[id0][2];
+        p[3] = vertPos[id1][0];
+        p[4] = vertPos[id1][1];
+        p[5] = vertPos[id1][2];
+        p[6] = vertPos[id2][0];
+        p[7] = vertPos[id2][1];
+        p[8] = vertPos[id2][2];
+        p += 9;
+
+        c[0] = numThreadIndex + 0;
+        c[1] = numThreadIndex + 1;
+        c[2] = numThreadIndex + 2;
+        c += 3;
+        numThreadIndex += 3;
+
+        m[0] = label[id0];
+        m += 1;
+      }
+    }
+  }
+#endif // TTK_ENABLE_OPENMP
+
+  this->printMsg(
+    "Wrote Boundaries", 1, localTimer.getElapsedTime(), this->threadNumber_);
+
+  return 0;
+}
+
+template <typename triangulationType>
+int ttk::MarchingTetrahedra::writeBoundariesDetailed_3D(
+  const unsigned char *const tetCases,
+  const size_t *numTriangles,
+  const unsigned long long *const scalars,
+  const triangulationType &triangulation) const {
+
+  ttk::Timer localTimer;
+
+  this->printMsg("Writing detailed boundaries", 0, 0, this->threadNumber_,
+                 ttk::debug::LineMode::REPLACE);
+
+  constexpr float diff = 0.02;
+  constexpr float d0 = 0.5 + diff;
+  constexpr float d1 = 0.5 - diff;
+  constexpr float dc = diff * 2;
+
+#ifdef TTK_ENABLE_OPENMP
+  if(threadNumber_ == 1) {
+#endif // TTK_ENABLE_OPENMP
+
+    output_points_->resize(9 * numTriangles[0]);
+    output_cells_connectivity_->resize(3 * numTriangles[0]);
+    output_cells_Labels_->resize(numTriangles[0]);
+    *output_numberOfPoints_ = 3 * numTriangles[0];
+    *output_numberOfCells_ = numTriangles[0];
+
+    auto &points = *output_points_;
+    auto &cellsConn = *output_cells_connectivity_;
+    auto &cellLabels = *output_cells_Labels_;
+
+    float *p = points.data();
+    SimplexId *c = cellsConn.data();
+    unsigned long long *m = cellLabels.data();
+    SimplexId cellIndex = 0;
+    const SimplexId numTets = triangulation.getNumberOfCells();
+
+    for(SimplexId tet = 0; tet < numTets; ++tet) {
+      if(!tetLookupIsMultiLabel[tetCases[tet]]) {
+        continue;
+      }
+
+      SimplexId vertices[4];
```

Comment:
Using zero-initialized `std::array` here instead of a C array should fix the remaining static analysis warning.